[WPE] Fix build error after 274090@main

dpino opened the pull request:
#### f7d58e8aeda0073f5cc181a78446cfdca892a7f8
<pre>
[WPE] Fix build error after 274090@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=265639">https://bugs.webkit.org/show_bug.cgi?id=265639</a>

Reviewed by Michael Catanzaro.

Only call &apos;WebCore::setScreenProperties&apos; if ENABLE(WPE_PLATFORM) is
defined.

* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::setScreenProperties):

Canonical link: <a href="https://commits.webkit.org/274140@main">https://commits.webkit.org/274140@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/183f7e5fba4dddf55d9fc3b1f8d74701596d2ccf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38084 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16995 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40353 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40631 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/33871 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19716 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14334 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/32157 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38658 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14354 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/33334 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12516 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/12447 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41907 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34575 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/34526 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/38330 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/13054 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/10722 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36523 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/14603 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8530 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13467 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/14053 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
```diff
@@ -62,6 +62,11 @@
+#if PLATFORM(GTK) || PLATFORM(WPE)
+#include <WebCore/PlatformScreen.h>
+#include <WebCore/ScreenProperties.h>
+#endif
```
Please don't guard header inclusion unless it's really necessary for the build. It makes future build failures more likely.

mcatanzaro commented on Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
```diff
@@ -259,7 +264,9 @@ void WebProcess::releaseSystemMallocMemory()
 #if PLATFORM(GTK) || PLATFORM(WPE)
 void WebProcess::setScreenProperties(const WebCore::ScreenProperties& properties)
 {
+#if PLATFORM(GTK) || (PLATFORM(WPE) && ENABLE(WPE_PLATFORM))
     WebCore::setScreenProperties(properties);
+#endif
```
This is correct, because the guards you use match the guards around PlatformScreen.cpp.

However, it might be better to change the guards around WebProcess::setScreenProperties instead, to force higher layers of the code to understand the conditions under which the function is available, rather than to just do nothing. That's what I would do, at least. Carlos Garcia might have a preference for one way or the other.

carlosgcampos replied:
This is ok to me.

dpino committed:
[WPE] Fix build error after 274090@main
https://bugs.webkit.org/show_bug.cgi?id=265639

Reviewed by Michael Catanzaro.

Only call 'WebCore::setScreenProperties' if ENABLE(WPE_PLATFORM) is
defined.

* Source/WebKit/WebProcess/glib/WebProcessGLib.cpp:
(WebKit::WebProcess::setScreenProperties):

Canonical link: https://commits.webkit.org/274140@main
